### PR TITLE
Is there a bug in the OHEMSampler?

### PR DESCRIPTION
**Pull request by yuzhj**
I got the error(TypeError: forward() missing 1 required positional argument: 'x_reg') when i used the OHEMSampler in the DoublehandRCNN, so I checkd the ohem_sampler.py and double_roi_head.py. 
line 27(double_roi_head.py) :
cls_score, bbox_pred = self.bbox_head(bbox_cls_feats, bbox_reg_feats)
line 37(ohem_sampler.py):
cls_score, _ = self.bbox_head(bbox_feats)
the arguments are different.
in the meantime, self.with_shared_head is in the double_roi_head.py but not in the ohem_sampler.py
so I did some fixes

### CODE REVIEW

File: mmdet/core/bbox/samplers/ohem_sampler.py [hunk 1]
```diff
@@ -21,20 +21,21 @@ def __init__(self,
...
-            self.bbox_roi_extractor = context.bbox_roi_extractor
-            self.bbox_head = context.bbox_head
+        self.context = context
+        if not hasattr(self.context, 'num_stages'):
+            self.bbox_head = self.context.bbox_head
         else:
-            self.bbox_roi_extractor = context.bbox_roi_extractor[
-                context.current_stage]
-            self.bbox_head = context.bbox_head[context.current_stage]
+            self.bbox_head = self.context.bbox_head[self.context.current_stage]
```
Comment (yhcao6) [inline mmdet/core/bbox/samplers/ohem_sampler.py:28]: There is something wrong with the OHEM usage in cascade rcnn. May I ask if you would like to fix that? If not, you can add an assertion here and I will fix it soon.

Comment (yuzhj) [reply mmdet/core/bbox/samplers/ohem_sampler.py:28]: Yes, I'd like to fix that, but I only have time on weekends

Comment (yuzhj) [reply mmdet/core/bbox/samplers/ohem_sampler.py:28]: > There is something wrong with the OHEM usage in cascade rcnn. May I ask if you would like to fix that? If not, you can add an assertion here and I will fix it soon.

Yes, I'd like to fix that, but I only have time on weekends.

Comment (yhcao6) [reply mmdet/core/bbox/samplers/ohem_sampler.py:28]: No problem

Comment (yuzhj) [reply mmdet/core/bbox/samplers/ohem_sampler.py:28]: > No problem

I've fixed the OHEM usage problem in cascade rcnn

Comment (yuzhj) [reply mmdet/core/bbox/samplers/ohem_sampler.py:28]: > No problem

I fixed the OHEM usage problem in cascade rcnn，please have a check

**Review by yhcao6 — APPROVED** [review]

**Pull request merged**